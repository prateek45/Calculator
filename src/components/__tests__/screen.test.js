import renderer from 'react-test-renderer';
import Screen from '../screen/screen';

const createNodeMock = el => {
    if (el.type === 'div') {
        const mockEl = document.createElement('div')
        mockEl.style.padding = '0'
        Object.defineProperty(mockEl, 'clientHeight', {
            get() {
                return 100
            },
            enumerable: true,
            configurable: true
        })
        Object.defineProperty(mockEl, 'clientWidth', {
            get() {
                return 300
            },
            enumerable: true,
            configurable: true
        })
        return mockEl
    }
    return null
}

test('Screen with particular value', () => {
    const component = renderer.create(
        <Screen value="53" />, { createNodeMock }
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Screen with particular value_2', () => {
    const component = renderer.create(
        <Screen value={5} />, { createNodeMock }
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});