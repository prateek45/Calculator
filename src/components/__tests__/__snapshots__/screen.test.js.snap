// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen with particular value 1`] = `
<div
  className="ant-card ant-card-bordered css-dev-only-do-not-override-1vtf12y"
  style={
    Object {
      "alignItems": "Center",
      "backgroundColor": "#4357692d",
      "marginBottom": "3%",
      "textAlign": "Center",
      "width": "100%",
    }
  }
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="screen"
      style={
        Object {
          "fontSize": 70,
        }
      }
    >
      <div
        style={
          Object {
            "display": "block",
            "whiteSpace": "nowrap",
          }
        }
      >
        53
      </div>
    </div>
  </div>
</div>
`;

exports[`Screen with particular value_2 1`] = `
<div
  className="ant-card ant-card-bordered css-dev-only-do-not-override-1vtf12y"
  style={
    Object {
      "alignItems": "Center",
      "backgroundColor": "#4357692d",
      "marginBottom": "3%",
      "textAlign": "Center",
      "width": "100%",
    }
  }
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="screen"
      style={
        Object {
          "fontSize": 70,
        }
      }
    >
      <div
        style={
          Object {
            "display": "block",
            "whiteSpace": "nowrap",
          }
        }
      >
        5
      </div>
    </div>
  </div>
</div>
`;
